integration/sawtooth_integration/tests/test_intkey_cli.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import unittest
+import time
+import logging
+import subprocess
+import shlex
+
+from sawtooth_integration.tests.integration_tools \
+    import wait_for_rest_apis, RestClient
+
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+
+
+# the appropriateness of these parameters depends on
+# the rate at which blocks are published
+WAIT_TIME = 3
+MINIMUM_BLOCK_COUNT = 3
+LOAD_COUNT = 3
+BATCH_COUNT = 10
+
+REST_API = 'rest_api:8080'
+URL = 'http://' + REST_API
+
+class TestIntkeyCLI(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        wait_for_rest_apis([REST_API])
+        cls.client = RestClient(URL)
+
+    def test_intkey_cli(self):
+        for _ in range(LOAD_COUNT):
+            _execute('intkey create_batch -v -c {} -o intkey.batch'.format(
+                BATCH_COUNT))
+            _execute('intkey load -v -f intkey.batch -U {}'.format(URL))
+            time.sleep(WAIT_TIME)
+
+        _execute('sawtooth block list --url {}'.format(URL))
+
+        # if the cli commands are working, expect at least
+        # MINIMUM_BLOCK_COUNT blocks to have been created
+        self.assertGreaterEqual(
```

Comment:
No. There's no way to verify the expected state, since the batches are created randomly. Another option would be to just run the commands without checking anything, but then it would be possible that the commands failed to do anything. The compromise is to verify that at least a few blocks were published, even if we can't say exactly how many.

I'm open to better ideas on how to test this effectively.